src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -188,16 +189,46 @@ public Result createNewExampleResultForSubmissionWithExampleSubmission(long subm
     }
 
     /**
-     * Save the feedback to the result with a workaround for Hibernate exceptions.
+     * Store the given feedback to the passed result (by replacing all existing feedback) with a workaround for Hibernate exceptions.
+     * <p>
+     * With ordered collections (like result and feedback here), we have to be very careful with the way we persist the objects in the database.
+     * We must first persist the child object without a relation to the parent object. Then, we recreate the association and persist the parent object.
      *
+     * If the result is not saved (shouldSave = false), the caller is responsible to save the result (which will persist the feedback changes as well)
+     *
+     * @param result           the result with should be saved with the given feedback
+     * @param feedbackList     new feedback items which replace the existing feedback
+     * @param shouldSave       whether the result should be saved or not
+     * @return the update (and potentially saved) result
+     */
+    public Result storeFeedbackInResult(@NotNull Result result, List<Feedback> feedbackList, boolean shouldSave) {
```

Comment:
```suggestion
    public Result storeFeedbacksInResult(@NotNull Result result, List<Feedback> feedbackList, boolean shouldSave) {
```

Reply:
Feedback does not really have a plural, so I keep the method name